src/leo_cache_tran.erl

```diff
@@ -0,0 +1,217 @@
+%%======================================================================
+%%
+%% LeoProject
+%%
+%% Copyright (c) 2013-2014 Rakuten, Inc.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%%======================================================================
+-module(leo_cache_tran).
+-author('Yosuke Hara').
+
+-behaviour(gen_server).
+
+-include("leo_cache.hrl").
+-include_lib("eunit/include/eunit.hrl").
+
+
+%% API
+-export([start_link/0,
+         stop/0]).
+
+-export([begin_tran/3,
+         wait_tran/2, wait_tran/3,
+         end_tran/2]).
+
+-export([init/1,
+         handle_call/3,
+         handle_cast/2,
+         handle_info/2,
+         terminate/2,
+         code_change/3]).
+
+-define(PROCESSDB, leo_cache_tran_proc_db).
+-define(REPLYDB, leo_cache_tran_reply_db).
+
+
+%%--------------------------------------------------------------------
+%% API
+%%--------------------------------------------------------------------
+%% Function: start_link() -> {ok,Pid} | ignore | {error,Error}
+%% Description: Starts the server
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+stop() ->
+    gen_server:call(?MODULE, stop).
+
+%% @doc
+-spec(begin_tran(pid(), atom(), binary()) ->
+             ok | {error, in_process}).
+begin_tran(Pid, Tbl, Key) ->
```

Comment:
No need to pass `Pid` because you can get the caller's pid from handle_call's argument `From`